docs/src/main/paradox/transactions.md

```diff
@@ -0,0 +1,72 @@
+# Transactions
+
+Kafka Transactions provide guarantees that messages processed in a consume->transform->produce workflow (consumed from a source topic, transformed, and produced to a destination topic) are processed exactly once or not at all.  This is achieved through coordination between the Kafka consumer group coordinator, transaction coordinator, and the consumer and producer clients used in the user application.  The Kafka producer marks messages that are consumed from the source topic as "committed" only once the transformed messages are successfully produced to the sink.   
+
+## Transactional Source
+
+The `Consumer.transactionalSource` emits a `ConsumerMessage.TransactionalMessage` which contains topic, partition, and offset information required by the producer during the commit process.  Unlike with `ConsumerMessage.CommittableMessage`, the user is not responsible for committing transactions, this is handled by `Producer.transactionalFlow` or `Producer.transactionalSink`.
+
+This source overrides the Kafka consumer property `isolation.level` to `read_committed`, so that only committed messages can be consumed.
+
+A consumer group ID must be provided.
+
+Only use this source if you have the intention to connect it to `Producer.transactionalFlow` or `Producer.transactionalSink`.
+
+## Transactional Sink and Flow
+
+The `Producer.transactionalSink` is similar to the `Consumer.commitableSink` in that messages will be automatically committed as part of a transaction.  The `Producer.transactionalSink` or `Producer.transactionalFlow` are required when connecting a consumer to a producer to achieve a transactional workflow.
+
+They override producer properties `enable.idempotence` to `true` and `max.in.flight.requests.per.connection` to `1` as required by the Kafka producer to enable transactions.
+
+A transactional ID must be defined and unique for each instance of the application.
```

Comment:
'transactional ID' - I'd certainly look for another name. I find it confusing with the ubiquitous  `transactionId`

Reply:
This is in reference to the `transactional.id` kafka producer config.

Reply:
OK if it's known in the "local dialect" :)